deepchem/models/torch_models/dtnn.py

```diff
@@ -0,0 +1,276 @@
+    >>> from deepchem.data import NumpyDataset
+    >>> model_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+    >>> input_file = os.path.join(model_dir, 'tests/assets/example_DTNN.mat')
+    >>> dataset = scipy_io.loadmat(input_file)
+    >>> X = dataset['X']
+    >>> y = dataset['T']
+    >>> w = np.ones_like(y)
+    >>> dataset = NumpyDataset(X, y, w, ids=None)
+    >>> n_tasks = y.shape[1]
+    >>> model = DTNNModel(n_tasks,
+    ...               n_embedding=20,
+    ...               n_distance=100,
+    ...               learning_rate=1.0,
+    ...               mode="regression")
+    >>> loss = model.fit(dataset, nb_epoch=250)
+    >>> pred = model.predict(dataset)
+
+    References
+    ----------
+    .. [1] Schütt, Kristof T., et al. "Quantum-chemical insights from deep
+        tensor neural networks." Nature communications 8.1 (2017): 1-8.
+
+    """
+
+    def __init__(self,
+                 n_tasks: int,
+                 n_embedding: int = 30,
+                 n_hidden: int = 100,
+                 n_distance: int = 100,
+                 distance_min: float = -1,
+                 distance_max: float = 18,
+                 output_activation: bool = True,
+                 mode: str = "regression",
+                 dropout: float = 0.0,
+                 **kwargs):
+        """
+        Parameters
+        ----------
+        n_tasks: int
+            Number of tasks
+        n_embedding: int (default 30)
+            Number of features per atom.
+        n_hidden: int (default 100)
+            Number of features for each molecule after DTNNStep
+        n_distance: int (default 100)
+            granularity of distance matrix
+            step size will be (distance_max-distance_min)/n_distance
+        distance_min: float (default -1)
+            minimum distance of atom pairs (in Angstrom)
+        distance_max: float (default = 18)
+            maximum distance of atom pairs (in Angstrom)
+        output_activation: bool (default True)
+            determines whether an activation function should be apply  to its output.
+        mode: str (default "regression")
+            Only "regression" is currently supported.
+        dropout: float (default 0.0)
+            the dropout probablity to use.
+
+        """
+        if dropout < 0 or dropout > 1:
+            raise ValueError("dropout probability has to be between 0 and 1, "
+                             "but got {}".format(dropout))
+        model = DTNN(n_tasks=n_tasks,
+                     n_embedding=n_embedding,
+                     n_hidden=n_hidden,
+                     n_distance=n_distance,
+                     distance_min=distance_min,
+                     distance_max=distance_max,
+                     output_activation=output_activation,
+                     mode=mode,
+                     dropout=dropout)
+        if mode not in ['regression']:
+            raise ValueError("Only 'regression' mode is currently supported")
```

Comment:
If there is only 1 mode, it should be removed as an arugment. you can always add a comment if there is a desire to add more later